test/e2e_node/device_plugin_test.go

```diff
@@ -942,7 +944,7 @@ func ensurePodContainerRestart(ctx context.Context, f *framework.Framework, podN
 		currentCount = p.Status.ContainerStatuses[0].RestartCount
 		framework.Logf("initial %v, current %v", initialCount, currentCount)
 		return currentCount > initialCount
-	}, 5*time.Minute, framework.Poll).Should(gomega.BeTrue())
+	}, 5*time.Minute, framework.Poll).Should(gomega.BeTrueBecause("expected current count %d to be greater than initial count %d", currentCount, initialCount))
```

Comment:
Instead of the result of the `currentCount > initialCount`, return `currentCount` and compare it with `gomega.Numerically(">", initialCount)`

No need to include both values in the message. `currentCount` doesn't need to be a variable outside of the anonymous function.

Reply:
What should I return inside the if block on line 941?

Reply:
Just return 0.

An even better solution would be to return the pod and then check with gomega that it has a `status.ContainerStatuses` where the element has a `restartCount` that is greater than before. But let's not get too fancy...